File: real-estate-management-system/c0721g2-srs-real-estate-fe/src/app/model/customer/customer.ts
import {AppUser} from '../user/app-user';
import {Image} from '../image/image';

export interface Customer {
  id?: string;
  address?: string;
  dateOfBirth?: string;
  email?: string;
  gender?: number;
  name?: string;
  idCard?: string;
  phoneNumber?: string;
  appUser?: AppUser;
  image?: Image;
  userName?: string;
  password?: string;
}

File: furama-resort-management-jsp-servlet/case_study_jsp_servlet/src/main/java/bean/Customer.java
package bean;

public class Customer extends Person {
    private String id;
    private CustomerType customerType;

    public Customer() {
    }

    public Customer(String name, String birthday, String gender, String personalID, String phoneNumber, String email, String address, String id, CustomerType customerType) {
        super(name, birthday, gender, personalID, phoneNumber, email, address);
        this.id = id;
        this.customerType = customerType;
    }

    public CustomerType getCustomerType() {
        return customerType;
    }

    public void setCustomerType(CustomerType customerType) {
        this.customerType = customerType;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}
